Minesweeper.py

```diff
@@ -44,132 +35,133 @@ def __init__(self):
             "The numbers say how many mines are adjacent to the tile cleared",
             "",
             "Press R to restart",
+            "Press ESCAPE to quit"
             "",
             "Press SPACE to start the game.",
         ]
 
-    def draw(self): # draws the tutorial screen
-        # DHB: using win as a global variable is a bad practice. It is better to send
-        # DHB: things like this into constructor args and store in member variables.
-        # DHB: When you start unit testing, global variables are a class of hell
-        # DHB: also, pretty sure you should only fill the window with the background color in the top-level draw function
+    def draw(self):
+        """ draws the tutorial screen using the text information from the init function
+        """
         win.fill(Color.BACKGROUND)
-        for i, instruction in enumerate(self.instructions):
+        for i, instruction in enumerate(self.message):
```

Comment:
the instruction variable name is not good here

Reply:
you're also still using 'win' as a global variable. Please inject it into the class instead. Global variables are truly evil constructs that need to be avoided at almost all costs. They destroy code maintainability, testability, refactorability, and predicatability.

Reply:
also as per prior comment elsewhere, try not to do inline calculations such as in the get_rect below. It leads to more bugs, less readable code.

Reply:
not sure how to do the last two so I'll do them later